compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/viewinterop/InteropContainer.skiko.kt

```diff
@@ -42,30 +43,53 @@ internal val LocalInteropContainer = staticCompositionLocalOf<InteropContainer>
 internal interface InteropContainer {
     val root: InteropViewGroup
     var rootModifier: TrackInteropPlacementModifierNode?
-    val interopViews: Set<InteropViewHolder>
-
-    fun placeInteropView(interopView: InteropViewHolder)
-    fun unplaceInteropView(interopView: InteropViewHolder)
-
-    // TODO: Should be the same as [Owner.onInteropViewLayoutChange]
-    fun changeInteropViewLayout(action: () -> Unit) {
-        action()
-    }
+    val snapshotObserver: SnapshotStateObserver
+
+    fun contains(holder: InteropViewHolder): Boolean
+
+    /**
+     * Calculates the proper index for the interop view in the container and issues a request to
+     * update the view hierarchy.
+     */
+    fun place(holder: InteropViewHolder)
+
+    /**
+     * Issues a request to remove the interop view from the hierarchy.
+     */
+    fun unplace(holder: InteropViewHolder)
+
+    /**
+     * Schedule an update to be performed on interop view. Platforms have their different strategy
+     * to align the updates with the rendering and threading requirements when modifying
+     * interop views.
+     *
+     * @param action The action to be performed. Could be layout change, or other visual updates
+     * to the view state, such as background, corner radius, etc.
+     */
+    fun scheduleUpdate(action: () -> Unit)
+
+    /**
+     * Callback to be invoked when the layout of the interop view changes to notify the system
+     * that something has changed.
+     */
+    fun onInteropViewLayoutChange(holder: InteropViewHolder)
```

Comment:
> Looking at a way how scheduleUpdate is written on Desktop, having this separate method is questionable again. I doubt that this method is required even on Android. Do you know if implementation of scheduleUpdate will be enough for Android?

Yes, I think so. If it's not enough, we will iterate on it.

> Since it's no-op for all platforms, let's comment it out with TODO (please restore TODO about Owner and merge with it)

Done.